services/agents/mongodb.go

```diff
@@ -25,11 +25,23 @@ import (
 	"github.com/percona/pmm/api/agentpb"
 	"github.com/percona/pmm/api/inventorypb"
 
+	"github.com/percona/pmm/version"
+
 	"github.com/percona/pmm-managed/models"
 )
 
 // mongodbExporterConfig returns desired configuration of mongodb_exporter process.
-func mongodbExporterConfig(service *models.Service, exporter *models.Agent, redactMode redactMode) *agentpb.SetStateRequest_AgentProcess {
+func mongodbExporterConfig(service *models.Service, exporter *models.Agent, redactMode redactMode, v *version.Parsed) *agentpb.SetStateRequest_AgentProcess {
+	// New MongoDB Exporter will be released with PMM agent v2.10.0
+	pmmNewVersion, _ := version.Parse("2.10.0")
+	if v.Less(pmmNewVersion) {
+		return mongodbExporterV1Config(service, exporter, redactMode)
```

Comment:
looks like the difference between these two function only in args and env variables, maybe it's better to return them, but keep common part in this function. What do you think?